src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/JourneyMapLocationController.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.controller
+
+import org.springframework.stereotype.Controller
+import org.springframework.ui.ModelMap
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.ModelAttribute
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.SessionAttributes
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.Supplier
+import uk.gov.justice.digital.hmpps.pecs.jpc.service.JourneyMapLocationService
+
+@Controller
+@SessionAttributes("supplier")
+class JourneyMapLocationController(private val service: JourneyMapLocationService) {
+
+  @GetMapping("/map-pickup-location/{journey-id}")
+  fun mapPickupLocation(@PathVariable("journey-id") journeyId: String,
+                        @ModelAttribute(name = "supplier") supplier: Supplier,
+                        model: ModelMap): String {
+
+    val agencies = service.findPickUpAndDropOffAgenciesForJourney(journeyId)
```

Comment:
So the other question I have is around how do we decided (if at all) whether mapping is allowed i.e. the nomis agency ids already have friendly names?